### ISSUE TITLE
An error message occurs when printing high resolution file, "Overload resolution failed"

### ISSUE THREAD

**Issue by winnerys** [stephanrauh/ngx-extended-pdf-viewer#530]
**Describe the bug**
I'm not sure there's resolution limitation when printing, I think this is resolution problem.
When the resolution exceeds a certain standard, the error message below appears when printing.

Uncaught TypeError: Failed to execute 'createObjectURL' on 'URL': Overload resolution failed.
at pdf_print_service.js:201
In my case, the PDF file's resolution is 5,080mm * 3,503mm.

And, I've already written the bug report on PDF.js.
but they say It's the limitation of a browser.
https://github.com/mozilla/pdf.js/issues/12547

**Version info**
- 5.1.0

**Desktop (please complete the following information):**
 - the latest version of Chrome

**To Reproduce**
1. Open the high resolution PDF file with a viewer and click the print icon.
  - in my case, 5,080mm * 3,503mm.(converted file from drawing)
2. The percentage of progress bar does not change from zero percent.

**Screenshots**
![image](https://user-images.githubusercontent.com/35720920/97644213-19c53e80-1a8d-11eb-9af5-54d1110f9eb6.png)

**Comment by stephanrauh** [comment]
Yes, the Mozilla guys are right. It first, I was confused because five by three millimeters isn't much - but I suppose it's five by three meters.

I didn't find it immediately, but somewhere on the showcase or in this bugtracker there's a list of the supported resolutions. It depends on your browser. If you're using Safari, the maximum resolution is 16 megapixels. Most other browsers allow larger files, but at the end of the day, the PDF file is painted on a canvas created by the browser, so there's always a limit. 

I'm not entirely sure if it really applies to printing, but did you try to reduce the attribute `[printResolution]`? Set it to something really low - such as 5 or 10 - to see if it helps. If it does, increase the resolution until the print look fine again. The default value is 150 DPI.

**Comment by stephanrauh** [comment]
TODO: document the browser limits on the showcase.

**Comment by stephanrauh** [comment]
Here's the list of maximum resolutions: https://www.npmjs.com/package/canvas-size#test-results

**Comment by winnerys** [comment]
> Here's the list of maximum resolutions: https://www.npmjs.com/package/canvas-size#test-results

Thank you. I'm looking for a solution. 
I think I can reply soon. 😃

**Comment by stephanrauh** [comment]
I've added the resolution check to print. If you're trying to print with extremely high resolution, the resolution is reduced to 95% of the browser limit. For some reason, that didn't work with every test document I tried. The more page the document has, the lower the maximum resolution. However, I hope 95% is a useful compromise between high quality and reasonable stability.

The enhancement has landed with version 7.0.0-beta.4.

Enjoy!
Stephan
